core/services/chainlink/config_general_secrets.go

```diff
@@ -41,3 +43,10 @@ func (g *generalConfig) PrometheusAuthToken() string {
 	}
 	return string(*g.secrets.Prometheus.AuthToken)
 }
+
+func (g *generalConfig) MercuryCredentials(credName string) (url, username, password string, err error) {
```

Comment:
Why not:

```
type MercuryCredentials struct {
  url string
  username string
  password string
}
```

Reply:
done.
it is difficult to add this struct in `chainlink` or `config/v2` packages bc of potential cyclic dependency. so i define it in the caller side `ocr2/models` package